optuna/integration/sklearn.py

```diff
@@ -730,7 +732,7 @@ def __init__(
         self.estimator = estimator
         self.max_iter = max_iter
         self.n_trials = n_trials
-        self.n_jobs = n_jobs
+        self.n_jobs = n_jobs if n_jobs else 1
```

Comment:
<img width="793" alt="image" src="https://user-images.githubusercontent.com/24619922/212594486-28746fc8-480a-4772-9b61-06de57f4181a.png">
Maybe we should also be consistent when used in a [joblib.parallel_backend](https://joblib.readthedocs.io/en/latest/parallel.html#joblib.parallel_backend) context?

Reply:
Yeah, this part is confusing. In `OptunaSearchCV`, `n_jobs` is for `study.optimize` so this argument is independent of `joblib.parallel_backend`.

Reply:
I see, thanks for explanation!